src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,276 @@
+
+/**
+ * Class that manages the reading from and 
+ * writing to local files in the local directory
+ */
+public class Storage {
+    private static final String TODO_IDENTIFIER = "T";
+    private static final String EVENT_IDENTIFIER = "E";
+    private static final String DEADLINE_IDENTIFIER = "D";
+
+    private static final String TODO_FULL_IDEN = typeTask.TODO.getTask();
+    private static final String EVENT_FULL_IDEN = typeTask.EVENT.getTask();
+    private static final String DEADLINE_FULL_IDEN = typeTask.DEADLINE.getTask();
+
+    private static final String INDICATOR_COMPLETE = "1";
+    private static final String INDICATOR_EVENT = "/at";
+    private static final String INDICATOR_DEADLINE = "/by";
+    // private static final String INDICATOR_INCOMPLETE = "0";
+
+    private final File fileSource;
+    private List<String> allStringTasks = new ArrayList<>();
+
+
+    /**
+     * Initializes a new storage class
+     * 
+     * @param filePath String indicating the path of 
+     * the local file to write to or read from.
+     */
+    public Storage(String filePath) {
+        this.fileSource = new File(filePath);
+        Path path = this.fileSource.toPath();
+        try {
+            Path parentPath = path.getParent();
+            Files.createDirectories(parentPath);
+            if (!Files.exists(path)) {
+                Files.createFile(path);
+            }
+            this.allStringTasks = Files.readAllLines(path);
+        } catch (IOException e) {
+            System.err.println("Error in path set!");
+        }
+    }
+
+    /**
+     * returns all the tasks stored in the local directory as an arraylist
+     * 
+     * @return Arraylist of Task object 
+     * @throws FileNotFoundException if local file directory does not exist
+     * @throws DukeException if either the Todo, Event, Deadline task is not 
+     * initialized because of a wrong input given by the user
+     */
+    protected ArrayList<Task> load() throws FileNotFoundException, DukeException {
+        HashMap<String, Boolean> stringTasks = new HashMap<>();
+        for (String tasks : allStringTasks) {
+            this.storeDiskStorageInputs(tasks, stringTasks);
+        }
+        return this.changeStorageToTasks(stringTasks);
+    }
+
+    private ArrayList<Task> changeStorageToTasks(
+            HashMap<String, Boolean> stringTasks) throws DukeException {
+        ArrayList<Task> finalOutputTasks = new ArrayList<>();
+        for (Map.Entry<String, Boolean> entry : stringTasks.entrySet()) {
+            String fullInstructions = entry.getKey();
+            String[] instructionsArray = fullInstructions.split(" ");
+            String crucialInstructions = String.join(" ",
+                    Arrays.copyOfRange(instructionsArray,
+                            1, instructionsArray.length));
+            String identity = instructionsArray[0];
+            if (identity.equals(TODO_FULL_IDEN)) {
+                Task ToDoTask;
```

Comment:
Maybe you want to install the checkstyle into your gradle package? I ran it on your files and it came up with 187 errors. Although most of them are minor things, there are multiple instances of coding standards violation like this where the variable names are not actually camelCase. Since your codebase is actually quite large automated checkstyle can probably help reduce such errors!